### ISSUE TITLE
Several flags have incorrect details

### ISSUE THREAD

**Issue by Poyo-SSB** [ppy/osu-web#3760]
Migrated from ppy/osu-resources#43. Since April, a few of the issues have been fixed. Here are the important ones which remain:

* Cambodia (`KH`): The central band should be red and the outer bands should be blue, not vice versa.
* Comoros (`KM`): As a symbol of Islam, the crescent (and stars) should probably not be distorted as they are.
* Sri Lanka (`LK`): The golden lion on the left side of the flag is completely missing.
* Mauritania (`MR`): The red stripes on the top and bottom are completely missing.
* Maldives (`MV`): The center is completely incorrect. It should not be blue, it should be green.

**Comment by PythonTryHard** [comment]
Alright, bodging

**Comment by arflyte** [comment]
![cambodia](https://user-images.githubusercontent.com/3632750/46713326-f0ac9800-cc90-11e8-91f7-d819dbecd519.png)
![comoros](https://user-images.githubusercontent.com/3632750/46713327-f0ac9800-cc90-11e8-9ad0-3916a4ab30dc.png)
![maldives](https://user-images.githubusercontent.com/3632750/46713328-f1452e80-cc90-11e8-9520-68c3ef302c3c.png)
![mauritania](https://user-images.githubusercontent.com/3632750/46713329-f1452e80-cc90-11e8-9c52-0a67bf8be491.png)
![sri lanka](https://user-images.githubusercontent.com/3632750/46713367-2fdae900-cc91-11e8-9fc0-609438ae2ca2.png)
![turkey](https://user-images.githubusercontent.com/3632750/46713331-f1ddc500-cc90-11e8-90ef-c3fdb0e776cb.png)

Fixed. Updated the repository.

**Comment by SurfChu85** [comment]
Would like to add Philippines (PH) as well, last time I checked the white triangle at the left only has a yellow circle instead of the "Three Stars and a Sun"

![current PH flag](https://puu.sh/BIBB6/44d5437b97.png)
![proposed PH flag](https://puu.sh/BIBvw/d93397ce28.png) (_just screenie'd discord's ph flag emoji_)

**Comment by Poyo-SSB** [comment]
I'll list the remaining non-critical issues here for posterity:

* Antigua and Barbuda (`AG`): The sun should be a sixteen-pointed star rather than a circle.
* Albania (`AL`): The double-headed eagle should be black instead of white.
* Australia (`AU`): The Southern Cross (Crux constellation) has five stars as represented on the [actual flag](https://upload.wikimedia.org/wikipedia/en/thumb/b/b9/Flag_of_Australia.svg/1280px-Flag_of_Australia.svg.png), and most of the six total stars on the flag are seven-pointed.
* Bolivia (`BO`): The coat of arms of Bolivia might be better represented than with a circle.
* Bhutan (`BT`): The bottom-right half of the flag should be orange, not red.
* China (`CN`): The four small stars should be pointed towards the larger star. The 2016 Summer Olympics at Rio [failed to adhere to this](http://time.com/4445373/rio-2016-olympics-china-wrong-flags/).
* Costa Rica (`CR`): The official ratio of the stripes is 1:1:2:1:1.
* Cuba (`CU`): The stripes should be equally sized.
* Cape Verde (`CV`): The official ratio of the stripes is 6:1:1:1:3, but having the central three stripes equally proportioned would be sufficient in my opinion. This might not work at this small a resolution, but there should also be ten stars.
* Spain (`ES`): Apparently, this simplification of the coat of arms could be considered offensive. I'm not Spanish, though, so take that with a grain of salt.
* Ethiopia (`ET`): The [emblem](https://en.wikipedia.org/wiki/Flag_of_Ethiopia#Emblem) in the center of the flag is quite heavily emphasized by the government.
* Fiji (`FJ`): The background of the flag should be a sky-blue rather than the navy seen in other similar flags.
* Micronesia (`FM`): The four stars should be pointing in each of the four cardinal directions.
* United Kingdom (`GB`): The red saltire of St. Patrick should be offset. This isn't really offensive, but it does look [silly to ignore](http://www.gizmodo.co.uk/2016/09/the-british-flag-at-the-olympics-was-totally-wrong-and-no-one-is-fixing-it-for-the-paralympics/).
* Guyana (`GY`): This representation of the flag is rather wonky. All the triangles should be vertically centered and the white/yellow triangles should extend to the right side of the flag.
* India (`IN`): As a spiritual symbol, the Ashoka Chakra (the 24-spoked wheel) should probably not be simplified.
* **Iran (`IR`):** Whoops, I missed this one back in April. The emblem in the middle of the flag is actually a stylized rendition of the text "Allah," so it should definitely not be simplified.
* South Korea (`KR`): The four black [trigrams](https://en.wikipedia.org/wiki/Bagua) represent national values, but have been reduced to meaningless double lines.
* Morocco (`MA`): As with Israel, the pentagram is a religious symbol (Seal of Solomon) and should not be simplified.
* Peru (`PR`): The coat of arms in the center is missing entirely.
* Philippines (`PH`): As mentioned above, the emblem on the triangle is perhaps oversimplified.
* Swaziland (`SZ`): The emblem in the center should probably not be simplified as drastically as it is.
* Thailand (`TH`): As with Costa Rica, the stripes' ratio is 1:1:2:1:1.
* Trinidad and Tobago (`TT`): The stripe is going the wrong way.
* Samoa (`WS`): As with Australia, the Southern Cross should have five stars.
* South Africa (`ZA`): Like Guyana, the flag is slightly wonky. The lines should be of equal width throughout.
* Zimbabwe (`ZW`): The triangle on the left should be white.

Here's a list of flags which currently stylize the star and/or crescent and maybe shouldn't:

* Azerbaijan (`AZ`)
* Libya (`LY`)
* Pakistan (`PK`) maybe?

**Comment by TPGPL** [comment]
wouldn't using twemoji opensource flags solve all the problems?

**Comment by Poyo-SSB** [comment]
Yes, but Twemoji also has a markedly different visual style from the rest of the osu! website. Personally, I'm quite fond of the pastel coloring that the flags have.

**Comment by Phippe** [comment]
The Mexican flag should probably use a darker green, a darker red and possibly a darker coat of arms too, because at first glance, it looks like the Italian one.

**Comment by arflyte** [comment]
Will look into expanding the colour palette for the flags.

**Comment by mrlmaokun** [comment]
Trinidad and Tobago flag is backwards, would love for it to be fixed.

**Comment by peppy** [comment]
By backwards, do you mean they are applied to the wrong country?

**Comment by arflyte** [comment]
Checked Trinidad and Tobago, and it's already fixed in the latest round of fixes.

**Comment by peppy** [comment]
Where are these fixes? Are they in a state we can deploy?

**Comment by arflyte** [comment]
It's all in #Resources/Flags

**Comment by peppy** [comment]
Which folder? Also are the filenames supposed to be country names instead of acronyms?

**Comment by peppy** [comment]
To clarify: there's

`Flags PNG`
`Flags PSD/flags_2-assets`
`New Flags PNG`
`New Flags PNG Small` (is this supposed to be @1x?)
zip files maybe containing the same, maybe different to above

almost comical levels of what

**Comment by arflyte** [comment]
There are only New Flags PNG and New Flags PNG Small (half the size, but it's not meant solely for @1x as I use it for some other small areas because when all of the designs were still done in Photoshop, it doesn't transform perfectly, because Adobe.)

Zip files are the resources for other designers to use. PSD, are the working files so I can edit/make new flags.

I'll just move the archives out of the folder.

**Comment by peppy** [comment]
Do you have a list of which flags have actually changed? And any idea why they have full country names not acronyms? We're going to have to go through and rename them to match again.

**Comment by arflyte** [comment]
That's all from the original file; all the layers were in full names. I can't remember if I manually changed everything one by one, or someone else converted it to acronyms.

**Comment by nanaya** [comment]
File names should be two lettered country code ("ISO 3166 code" section in Wikipedia), not the three letters version (which is mixed in in the 2x directory) ಠ_ಠ

The flag for TT (Trinidad and Tobago) is still going the wrong way. It should go like this: `\` and not like this: `/`.

Current: ![](https://osu.ppy.sh/images/flags/TT.png)

Correct: ![](https://s.myconan.net/2020-06/120px-Flag_of_Trinidad_and_Tobago.png)

**Comment by arflyte** [comment]
![Trinidad and Tobago](https://user-images.githubusercontent.com/3632750/83508262-9ebf2280-a504-11ea-8da4-3dc7d29992c1.png)
This is the one that's fixed. It's in the folder.

**Comment by nanaya** [comment]
Hmm that was confusing. Also I guess someone will have to rename them again.

**Comment by Trigonoculus** [comment]
![image](https://user-images.githubusercontent.com/20501245/88884257-69af2200-d260-11ea-8755-adf820f42186.png)

[The flag of Guinea - GN.png](/public/images/flags/GN.png) is wrongly arranged. The red color should be at the furthest left, with the yellow color being in the middle.

For reference, please see https://en.wikipedia.org/wiki/Flag_of_Guinea

**Comment by Joehuu** [comment]
https://www.flaticon.com/packs/international-flags-6 seems to have the same design language and still looks good on small sizes without simplifying/modifying the flag (that ruins it).

Just want to push this discussion more towards an icon pack instead of making your own flags that, to be honest, have a lot of overlooked adjustments to the actual flag.

**Comment by bdach** [comment]
I'm no vexillologist by any means of the word, but those linked ones aren't all perfect either. To list a few examples, that flag pack also takes some artistic license to simplify the flags of [Egypt](https://www.flaticon.com/free-icon/egypt_330527), [Lebanon](https://www.flaticon.com/free-icon/lebanon_330667), [Cambodia](https://www.flaticon.com/free-icon/cambodia_330526?related_id=330526&origin=pack), or [Turkmenistan](https://www.flaticon.com/free-icon/turkmenistan_330456), as a few loose immediate examples.

**Comment by Kotoki1337** [comment]
> https://github.com/ppy/osu-web/issues/3760#issuecomment-428825836
> -  China (`CN`): The four small stars should be pointed towards the larger star. The 2016 Summer Olympics at Rio [failed to adhere to this](http://time.com/4445373/rio-2016-olympics-china-wrong-flags/).

I think we need to talk to solve this issue, the censorship in the mainland is strict, now the video website found the wrong flag in the osu! and started to delete the osu! video with the wrong flag. Because it violates the ` Law on the National Flag of the People's Republic of China`.

### About the video
![The video has been deleted](https://user-images.githubusercontent.com/35394151/105465898-7c198e00-5cce-11eb-91d9-be2c4e315d62.png)

![Talk with the website support](https://user-images.githubusercontent.com/35394151/105467462-b08e4980-5cd0-11eb-823a-481088ffb423.png)

We are worried that someone will report the video maliciously because almost all tournament videos and Chinese player replays contain the flag of osu!
We once appeared that one of our organization accounts that uploaded osu! was permanently banned because the live video of a top player accidentally included something prohibited by China.

Although `Hong Kong` flag (even if we according to the Olympic standards specify `Hong Kong, China`, some OWC2019, 2020 videos are still deleted) and the `Republic of China` flag internationally representing `Taiwan` will also be deleted (so we using `Chinese Taipei` represent `Taiwan` in the Olympic standard to bypass censorship), this makes very difficult for Chinese ppl to upload tournament videos or player replay in China mainland. People need to overlay the wrong flag of the entire video to bypass the censorship. Of course, there is no problem with the flags of Hong Kong and Taiwan, and no need modification. But at least we should fix the flag issue of China.

The previous video may not be saved, but we hope that the wrong flag should be fixed.

---
## The wrong flag in osu!
![current China flag in osu!](https://raw.githubusercontent.com/ppy/osu-web/master/public/images/flags/CN.png)

## The flag of China Standardization
[Flag of China : Construction details, sizes and colors (Wikipedia EN)](https://en.wikipedia.org/wiki/Flag_of_China#Construction_details,_sizes_and_colors)
[中华人民共和国国旗：规格 (Wikipedia ZH)](https://zh.wikipedia.org/wiki/%E4%B8%AD%E5%8D%8E%E4%BA%BA%E6%B0%91%E5%85%B1%E5%92%8C%E5%9B%BD%E5%9B%BD%E6%97%97#%E8%A7%84%E6%A0%BC)
![](https://upload.wikimedia.org/wikipedia/commons/a/ab/Construction_sheet_of_Flag_of_the_People%27s_Republic_of_China.svg)
- The flag is split into 4 sections, with the top hoist part of the flag being a grid of 15 by 10 units.
- The center of the biggest star is placed at 5 units from the hoist and 5 units from the top of the flag; the diameter of the biggest star's circumscribed circle is 6 units.
- Of the 4 smaller stars, the first one is centered 2 units from the top of the flag, 10 units from the hoist; the second one is centered 4 units from the top of the flag and 12 units from the hoist; the third one is centered 7 units from the top of the flag and 12 units from the hoist; the fourth one is centered 9 units from the top of the flag and 10 units from the hoist.
- The diameter of each small star's circumscribed circle is 2 units. **Each of the top points of the 4 smaller stars are rotated such that they point towards the center of the larger star.**

## TL;DR
**Each of the top points of the 4 smaller stars are rotated such that they point towards the center of the larger star.**

## Solution I think
- Use the [China icon](https://www.flaticon.com/free-icon/china_330651) in the [International flags Icon Pack](https://www.flaticon.com/packs/international-flags-6) https://github.com/ppy/osu-web/issues/3760#issuecomment-764860485. As stated in the comment, the design language is the same, can be used for commercial, and the flag is correct.
- Directly modify the existing China flag of osu!, The one top points of the four small stars are aligned with the center of the big star.


---
### Ref
- [National Flag Law of the PRC (Unofficial translation)](https://www.chinalawtranslate.com/en/flaglaw2020/)

**Comment by peppy** [comment]
@arflyte

**Comment by arflyte** [comment]
This was brought up before and has been fixed in the repo. Seems like one of the star in the latest update is still misaligned. Uploaded the latest one to the repo.

Moving forward, maybe we should switch to twemoji?

**Comment by Kotoki1337** [comment]
> This was brought up before and has been fixed in the repo. Seems like one of the star in the latest update is still misaligned. Uploaded the latest one to the repo.
> 
> Moving forward, maybe we should switch to twemoji?

uh which repo been fixed? osu-web, osu-wiki and current website, lazer flag are wrong.

**Comment by arflyte** [comment]
I was referring to the internal design repository.

**Comment by Kotoki1337** [comment]
oh nvm, sound great.

**Comment by Cafe17951** [comment]
The dimensions of the Costa Rican flag are incorrect, The blue stripe is too large equally the white stripe is too small when these two stripes should be the same size.

The red stripe is equivalent to size by joining a blue and white stripe together.

![Screenshot_20201214-145331_Chrome](https://user-images.githubusercontent.com/38241013/105652587-39e37e80-5e7f-11eb-8b55-ad85ede339b7.jpg)

**Comment by Joehuu** [comment]
Costa Rica is already mentioned in https://github.com/ppy/osu-web/issues/3760#issuecomment-428825836.

**Comment by Kaplankook** [comment]
Laos (LA): Stripes should be 1:2:1 instead of 1:1:1 
Moldova (MA): Eagle in the middle doesn't look anything like this
Serbia (RS): There should be a Shield instead of Eagle + the Eagle isn't yellow but white

**Comment by peppy** [comment]
Another one pointed out [here](https://twitter.com/DaBluBerryPie/status/1415403691977420801) is Albania's symbol being the wrong colour.

Would probably be good to get some kind of conclusion on this issue, whether that involves switching to a new more correct flag set or fixing the remaining outstanding issues.

**Comment by nanaya** [comment]
hi @arflyte

**Comment by arflyte** [comment]
I think it's best to just switch to twemoji. It's used almost everywhere, the size is almost the same, and probably they have vexillologists to help them check. The ones that we have probably missed many more, just that we haven't discovered it yet. Switching to twemoji would probably end all of the flag problems we're having now.

**Comment by peppy** [comment]
I guess the only issue with that is that we are switching away from our pastel colour palette, which is a bit of a shame. But I guess if no one is willing to put in the effort to fix the remaining issues that's how it will be..

**Comment by arflyte** [comment]
I really really love the pastel colour palette too, but there were some concerns saying that "we are changing the colours, and disrespecting their flags". Some countries are pretty strict with the colours :(

It's a touchy subject, and I'm not even sure if creative license applies to flags; personal projects maybe, but not for something as huge as osu! i think?
